internal/wire/parse.go

```diff
@@ -860,25 +894,44 @@ func processBind(fset *token.FileSet, info *types.Info, call *ast.CallExpr) (*If
 	// Assumes that call.Fun is wire.Bind.
 
 	if len(call.Args) != 2 {
-		return nil, notePosition(fset.Position(call.Pos()), errors.New("call to Bind takes exactly two arguments"))
+		return nil, notePosition(fset.Position(call.Pos()),
+			errors.New("call to Bind takes exactly two arguments"))
 	}
 	// TODO(light): Verify that arguments are simple expressions.
 	ifaceArgType := info.TypeOf(call.Args[0])
 	ifacePtr, ok := ifaceArgType.(*types.Pointer)
 	if !ok {
-		return nil, notePosition(fset.Position(call.Pos()), fmt.Errorf("first argument to Bind must be a pointer to an interface type; found %s", types.TypeString(ifaceArgType, nil)))
+		return nil, notePosition(fset.Position(call.Pos()),
+			fmt.Errorf("first argument to Bind must be a pointer to an interface type; found %s", types.TypeString(ifaceArgType, nil)))
 	}
 	iface := ifacePtr.Elem()
 	methodSet, ok := iface.Underlying().(*types.Interface)
 	if !ok {
-		return nil, notePosition(fset.Position(call.Pos()), fmt.Errorf("first argument to Bind must be a pointer to an interface type; found %s", types.TypeString(ifaceArgType, nil)))
+		return nil, notePosition(fset.Position(call.Pos()),
+			fmt.Errorf("first argument to Bind must be a pointer to an interface type; found %s", types.TypeString(ifaceArgType, nil)))
 	}
+
 	provided := info.TypeOf(call.Args[1])
+	if bindToUsePointer {
+		providedPtr, ok := provided.(*types.Pointer)
+		if !ok {
+			return nil, notePosition(fset.Position(call.Args[0].Pos()),
+				fmt.Errorf("second argument to Bind must be a pointer or a pointer to a pointer; found %s", types.TypeString(provided, nil)))
+		}
+		provided = providedPtr.Elem()
+	} else {
+		fmt.Fprintf(os.Stderr,
```

Comment:
I'm worried that we're not giving the user a clear enough call to action here: if the user's use a pointer in the second argument to `Bind` without updating the package, it will have the wrong behavior. In fact, the user is using the tool correctly but it will break when they update the Wire library package.

IMO, this situation is nuanced enough to communicate that I think it's okay to just let it break loudly when they upgrade their Wire marker functions. If they are using the old way with the old library, that is okay.

Reply:
Discussed offline, we don't want the error message when the user is using the new command with the old wire library. When user upgrades their wire library, their existing `Bind` calls will behave differently and fail later during the generation phase. We should document and let them know how to fix it.